src/test/java/com/artipie/rpm/RpmITCase.java

```diff
@@ -23,201 +23,77 @@
-        this.startArtipie(anonymous);
-        this.prepareRpmRepository();
-        this.prepareContainer(anonymous);
-        MatcherAssert.assertThat(
-            "Installs 'time' package",
-            this.yumExec("install"),
-            new StringContainsInOrder(new ListOf<>("time-1.7-45.el7.x86_64", "Complete!"))
-        );
-    }
-
-    @AfterEach
-    void close() {
-        this.server.stop();
-        if (this.cntn != null) {
-            this.cntn.close();
-        }
-    }
-
-    private String yumExec(final String action) throws Exception {
-        return this.cntn.execStdout(
-            "yum", "-y", "repo-pkgs", "example", action
-        );
-    }
-
-    private void prepareContainer(final boolean anonymous)
-        throws Exception {
-        Testcontainers.exposeHostPorts(this.port);
-        final Path setting = this.tmp.resolve("example.repo");
-        this.tmp.resolve("example.repo").toFile().createNewFile();
-        Files.write(
-            setting,
-            new ListOf<>(
-                "[example]",
+        this.containers.putBinaryToClient(
+            String.join(
+                "\n", "[example]",
                 "name=Example Repository",
-                String.format(
-                    "baseurl=%s", new RepositoryUrl(this.port, RpmITCase.REPO).string(anonymous)
-                ),
+                "baseurl=http://artipie:8080/my-rpm",
                 "enabled=1",
                 "gpgcheck=0"
-            )
+            ).getBytes(),
+            "/etc/yum.repos.d/example.repo"
         );
-        this.cntn = new TestContainer("centos:centos8", this.tmp);
-        this.cntn.start(this.port);
-        this.cntn.execStdout("mv", "/home/example.repo", "/etc/yum.repos.d/");
     }
 
-    private void prepareRpmRepository() {
-        final Storage storage = new FileStorage(this.tmp);
-        new TestResource("rpm/time-1.7-45.el7.x86_64.rpm")
-            .saveTo(storage, new Key.From(RpmITCase.REPO, "time-1.7-45.el7.x86_64.rpm"));
-        new Rpm(
-            storage,
-            new RepoConfig.Simple(Digest.SHA256, new NamingPolicy.HashPrefixed(Digest.SHA1), true)
-        ).batchUpdate(new Key.From(RpmITCase.REPO)).blockingAwait();
+    @Test
+    void addsRpmAndCreatesRepodata() throws Exception {
+        this.containers.assertExec(
+            "Failed to upload rpm package",
+            new MavenITCase.ContainerResultMatcher(),
+            "curl", "http://artipie:8080/my-rpm/time-1.7-45.el7.x86_64.rpm",
+            "--upload-file", "/w/time-1.7-45.el7.x86_64.rpm"
+        );
+        this.containers.assertExec(
+            "Failed to install time package",
+            new MavenITCase.ContainerResultMatcher(
+                new IsEqual<>(0),
+                new StringContainsInOrder(new ListOf<>("time-1.7-45.el7.x86_64", "Complete!"))
+            ),
+            "yum", "-y", "repo-pkgs", "example", "install"
+        );
```

Comment:
@olenagerasimova in this method also installation is checked, therefore probably to rename this method a bit